accounting/src/delta/delta_data_collection/tests/mod.rs

```diff
@@ -0,0 +1,322 @@
+// Copyright (c) 2021-2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::*;
+
+mod delta_delta_delta_delta_tests;
+mod delta_delta_delta_tests;
+mod delta_delta_undo_tests;
+mod delta_delta_undo_undo_tests;
+
+#[test]
+#[rustfmt::skip]
+fn test_combine_deltas() {
+    use DataDelta::{Create, Delete, Modify};
+
+    assert_eq!(combine_delta_data(&Create('a'), Create('b')),      Err(Error::DeltaDataCreatedMultipleTimes));
+    assert_eq!(combine_delta_data(&Create('a'), Modify('a', 'b')), Ok(Some(Create('b'))));
+    assert_eq!(combine_delta_data(&Create('a'), Delete('a')),      Ok(None));
+
+    assert_eq!(combine_delta_data(&Modify('a', 'b'), Create('c')),      Err(Error::DeltaDataCreatedMultipleTimes));
+    assert_eq!(combine_delta_data(&Modify('a', 'b'), Modify('c', 'd')), Ok(Some(Modify('a', 'd'))));
```

Comment:
Are we sure this is valid, to modify a -> b then c -> d? Shouldn't it be a -> b then b -> c? Is allowing this OK for some reason I'm not noticing?